vars/main.yml

```diff
@@ -4,3 +4,12 @@ __timesync_chrony_version: "{{
 __timesync_ntp_version: "{{
   ansible_facts.packages['ntp'][0].version|default('0')
   if 'ntp' in ansible_facts.packages else '0' }}"
+
+# ansible_facts required by the role
+__timesync_required_facts:
+  - distribution
+  - distribution_major_version
+  - distribution_version
+  - os_family
+  - packages
+  - services
```

Comment:
I don't think packages and services should be in this list - they are not required to determine which packages to install - there are separate `package_facts:` and `service_facts:` modules for those.